core/include/core/versionchecker.h

```diff
@@ -12,24 +12,57 @@ class SCOPY_CORE_EXPORT VersionCache : public QObject
 	Q_OBJECT
 
 protected:
-	VersionCache(QObject *parent = nullptr);
+	explicit VersionCache(QObject *parent = nullptr);
 	void init();
-	~VersionCache();
+	~VersionCache() override;
 
 public:
 	VersionCache(VersionCache &other) = delete;
 	void operator=(const VersionCache &) = delete;
 
 	static VersionCache *GetInstance();
+
+	/**
+	 * @brief Pull the json file with the current version from m_url and save it in the m_cache variable.
+	 * */
 	void updateCache();
```

Comment:
Review the interface of the function.
Ideally, besides singleton boilerplate, there should be only one function
checkVersion(QObject, QCallback*(QJsonDocument) )